File: frontend/src/types/datashow.ts
export interface OverviewCard {
  label: string;
  value: string;
  change: string;
  trend_up: boolean | null;
  color: string;
  trend_data: number[];
}

export interface RiskTrendBar {
  month: string;
  high: number;
  medium: number;
  low: number;
}

export interface IndustrySegment {
  label: string;
  value: number;
  color: string;
  percentage: string;
}

export interface AlertItem {
  company: string;
  event: string;
  time: string;
  level: "high" | "medium" | "low" | "info";
}

export interface InsightItem {
  icon: string;
  title: string;
  description: string;
  type: string;
}

export interface OverviewResponse {
  cards: OverviewCard[];
  risk_trend: RiskTrendBar[];
  industry_distribution: IndustrySegment[];
  alerts: AlertItem[];
  insights: InsightItem[];
}

// NL Query → Chart
export interface NLChartResponse {
  chart_type: string;
  title: string;
  description: string;
  echarts_option: Record<string, unknown>;
}

// Relationship Graph
export interface GraphNode {
  id: string;
  name: string;
  category: string;
  value?: number;
  symbol_size: number;
}

export interface GraphEdge {
  source: string;
  target: string;
  label: string;
  value?: number;
}

export interface GraphResponse {
  nodes: GraphNode[];
  edges: GraphEdge[];
  categories: string[];
}

// Trend comparison
export interface TrendSeries {
  name: string;
  data: number[];
  color: string;
}

export interface TrendResponse {
  months: string[];
  series: TrendSeries[];
}

File: backend/app/schemas/datashow.py
"""Data visualization / dashboard schemas."""

from pydantic import BaseModel
from typing import Optional


class OverviewCard(BaseModel):
    label: str
    value: str
    change: str
    trend_up: Optional[bool]
    color: str
    trend_data: list[float]


class RiskTrendBar(BaseModel):
    month: str
    high: int
    medium: int
    low: int


class IndustrySegment(BaseModel):
    label: str
    value: int
    color: str
    percentage: str


class AlertItem(BaseModel):
    company: str
    event: str
    time: str
    level: str


class InsightItem(BaseModel):
    icon: str
    title: str
    description: str
    type: str


class OverviewResponse(BaseModel):
    cards: list[OverviewCard]
    risk_trend: list[RiskTrendBar]
    industry_distribution: list[IndustrySegment]
    alerts: list[AlertItem]
    insights: list[InsightItem]


# ---------------------------------------------------------------------------
# NL Query → Chart
# ---------------------------------------------------------------------------

class NLQueryRequest(BaseModel):
    query: str


class NLChartResponse(BaseModel):
    chart_type: str  # bar / line / pie / radar / scatter
    title: str
    description: str
    echarts_option: dict  # Full ECharts option, directly renderable


# ---------------------------------------------------------------------------
# Relationship Graph
# ---------------------------------------------------------------------------

class GraphNode(BaseModel):
    id: str
    name: str
    category: str
    value: Optional[float] = None
    symbol_size: float = 30


class GraphEdge(BaseModel):
    source: str
    target: str
    label: str
    value: Optional[float] = None


class GraphResponse(BaseModel):
    nodes: list[GraphNode]
    edges: list[GraphEdge]
    categories: list[str]
